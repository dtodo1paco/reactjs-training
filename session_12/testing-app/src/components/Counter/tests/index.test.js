import React from 'react';
import { shallow } from 'enzyme';
import Component from '../index';

describe(`${Component.displayName} basics`, () => {
  it("should render correctly", () => {
    const wrapper = shallow(<Component />);
  });
  it("should match snapshot", () => {
    const wrapper = shallow(<Component />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
describe(`${Component.displayName} structure`, () => {
  it("should have a button", () => {
    const wrapper = shallow(<Component />);
    expect(wrapper.find('button').length).toEqual(1);
    expect(wrapper.find('h2').length).toEqual(1);
  });
});
describe(`${Component.displayName} behavior`, () => {
  it("should handle button click and update state", () => {
    const wrapper = shallow(<Component />);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(wrapper.state('count')).toEqual(2);
  });
  it("should render using state", () => {
    const wrapper = shallow(<Component />);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(wrapper.find('h2').text()).toEqual('Number of clicks: 2');
  });
});