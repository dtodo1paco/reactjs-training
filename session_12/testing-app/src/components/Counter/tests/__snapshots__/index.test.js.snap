// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter basics should match snapshot 1`] = `
Array [
  <div
    className="counter"
  >
    <h2>
      Number of clicks: 
      1
    </h2>
    <p>
       This is my component. Please click it!
    </p>
    <button
      onClick={[Function]}
    >
      Increment!
    </button>
  </div>,
]
`;
